File: selfstudy/golang/words_max.go
/*
task 292:
Words - Max
You need to find the maximum value among word lengths
*/

package main

import (
	"fmt"
	"reflect"
)

func HighestWords(lengths []int) int {
	if len(lengths) == 0 { return 0 }
	m := lengths[0]
	for _, v := range lengths {
		if v > m { m = v }
	}
	return m
}

func main() {
	tests := []struct {
		input []int
		expected int
	}{
		{[]int{10, 20, 30}, HighestWords([]int{10, 20, 30})},
		{[]int{50, 60, 70}, HighestWords([]int{50, 60, 70})},
		{[]int{1, 2, 3}, HighestWords([]int{1, 2, 3})},
		{[]int{100, 200}, HighestWords([]int{100, 200})},
		{[]int{}, HighestWords([]int{})},
	}
	
	for idx, t := range tests {
		res := HighestWords(t.input)
		status := "FAIL"
		if reflect.DeepEqual(res, t.expected) {
			status = "PASS"
		}
		fmt.Printf("Test %d: %s\n", idx+1, status)
	}
}

// works i think

File: selfstudy/js/words_max.js
/*
task 92:
Words - Max
You need to find the maximum value among word lengths
*/

function highest_words(lengths) {
  return lengths.length ? Math.max(...lengths) : 0;
}

// Tests
const tests = [
  { input: [10, 20, 30], expected: highest_words([10, 20, 30]) },
  { input: [50, 60, 70], expected: highest_words([50, 60, 70]) },
  { input: [1, 2, 3], expected: highest_words([1, 2, 3]) },
  { input: [100, 200], expected: highest_words([100, 200]) },
  { input: [], expected: highest_words([]) }
];

if (process.argv.includes('mocha')) {
  const assert = require('assert');
  describe('highest_words', () => {
    tests.forEach((t, idx) => {
      it(`test case ${idx+1}`, () => {
        assert.deepStrictEqual(highest_words(t.input), t.expected);
      });
    });
  });
} else {
  tests.forEach((t, idx) => {
    console.log(`Test ${idx+1}:`, highest_words(t.input) === t.expected || JSON.stringify(highest_words(t.input)) === JSON.stringify(t.expected) ? "PASS" : "FAIL");
  });
}

// works i think
